### ISSUE TITLE
(🐞) New union syntax inconsistent with old union syntax, now it contains `Any`

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#12224]
```py
from typing import Union
class A:
    def __init__(self, a): ...
    
Test1 = str | A  # error
Test2 = Union[str, A]
```
`main.py:5: error: Expression type contains "Any" (has type "Type[A]")`
But no error on the last line

https://mypy-play.net/?mypy=latest&python=3.10&flags=disallow-any-expr&gist=dd6dcaf336fb58d4afc3310200259630

**Comment by ktbarrett** [comment]
I think the issue is that it is not considering the type `A` as typed since the `__init__` is not typed.

The following passes without issue.
```python
from typing import Union

class A:
    def __init__(self, a: int) -> None: ...
    
Test1 = str | A
Test2 = Union[str, A]
```

**Comment by erictraut** [comment]
This bug was apparently fixed in mypy 1.0.0. I think it can be closed.

**Comment by hauntsaninja** [comment]
Fixed in #14181
